### ISSUE TITLE
0.24.0-insiders - Syntax colorization broken on some themes

### ISSUE THREAD

**Issue by bobbrow** [microsoft/vscode-cpptools#3779]
**Type: LanguageService**

In 0.24.0-insiders, there are some themes rendering the syntax highlighting incorrectly.  if this affects your theme, please add a comment to this issue.  We are working out which ones were broken with our syntax highlighting change.

To disable the new syntax highlighting, set `"C_Cpp.enhancedColorization": "Disabled"` and you will get the old behavior again.

Known themes with this issue:
* Tomorrow Night Blue
* High Contrast

**Comment by manasjoshi14** [comment]
Affects the Monokai theme.
"C_Cpp.enhancedColorization": "Disabled" resolved it.

**Comment by Pigzaum** [comment]
This is affecting the Atom One Dark when using `"editor.renderWhitespace": "all"`.
The white spaces inside a comment are highlighted in the same color as words, for example:

![comment](https://user-images.githubusercontent.com/14007530/59553219-67a8d780-8f68-11e9-8f38-3191d1d20444.png)

`"C_Cpp.enhancedColorization": "Disabled"` resolved it as well.

**Comment by TomiCode** [comment]
Also affects the [vscode-ayu](https://github.com/ayu-theme/vscode-ayu) themes, example below. Of course disabling the mentioned option fixes the issue.
<img width="523" alt="Zrzut ekranu 2019-06-16 o 11 34 05" src="https://user-images.githubusercontent.com/1648325/59562251-b323ba00-902a-11e9-9787-c3a04a8c9be0.png">

**Comment by tomasiser** [comment]
@Pigzaum 
Whitespace character rendering in comments has wrong colors also in the default theme:
![image](https://user-images.githubusercontent.com/10374559/59569179-37575b00-9086-11e9-80c6-dc4f018208d5.png)
I believe it should have the same color as everywhere else.
Furthermore, changing the color with:
```
"workbench.colorCustomizations": {
    "editorWhitespace.foreground": "#ffffff22"
},
```
does not affect the comments.

The same problem occurs with strings:
![image](https://user-images.githubusercontent.com/10374559/59569269-2ce99100-9087-11e9-9999-4198de5f62e6.png)
Again, the whitespaces are orange instead of gray.

**Comment by Colengms** [comment]
Please upvote the following issue: https://github.com/microsoft/vscode/issues/49462

**Comment by sean-mcmanus** [comment]
One potential "fix" for the the editor.renderWhitespace issue is:
1. If the user has the "all" setting set, then disable enhancedColorization by default.
2. If the user has the "boundary" setting set, then exclude > 1 whitespace from decorations.

**Comment by sean-mcmanus** [comment]
Fixed with 0.24.0-insiders2. Set the "C_Cpp.updateChannel" setting to "Insiders" to auto-install it.
